lib/logstash/outputs/elasticsearch/http_client/pool.rb

```diff
@@ -225,9 +229,16 @@ def healthcheck!
       # Try to keep locking granularity low such that we don't affect IO...
       @state_mutex.synchronize { @url_info.select {|url,meta| meta[:state] != :alive } }.each do |url,meta|
         begin
-          logger.info("Running health check to see if an Elasticsearch connection is working",
-                      url: url, healthcheck_path: healthcheck_path)
-          response = perform_request_to_url(url, :head, healthcheck_path)
+          if @absolute_healthcheck_path
+            absolute_healthcheck_safe_url = ::LogStash::Util::SafeURI.new(healthcheck_path)
+            logger.info("Running health check to see if an Elasticsearch connection is working",
+                        absolute_healthcheck_url: absolute_healthcheck_safe_url.sanitized)
+            response = perform_request_to_url(healthcheck_path, :head, ROOT_URI_PATH)
```

Comment:
At this point everything uses the SafeURI. We only unwrap it briefly in the ManticoreAdapter *just* before passing it in to Manticore (which does not understand SafeURI)